src/Elmish/Hooks/Type.js

```diff
@@ -4,15 +4,14 @@ const uniqueNameFromCurrentCallStack_ = ({ trace }) => ({ skipFrames, prefix })
   const stack = new Error().stack
   const stackLines = stackTraceParser.parse(stack)
   const hookCallSite = stackLines[skipFrames]
-  const file = cleanName(hookCallSite.file)
   const methodName = cleanName(hookCallSite.methodName)
   if (trace) {
     console.log('Hook Call Site:')
     console.log(hookCallSite)
     console.log('Full Stack Trace:')
     console.log(stack)
   }
-  return `${prefix}_${file}_${methodName}_${hookCallSite.lineNumber}_${hookCallSite.column}`
+  return `${prefix}_${methodName}_${hookCallSite.lineNumber}_${hookCallSite.column}`
```

Comment:
Shouldn't `file` be present here as well? Or is it always guaranteed to be present in `methodName`?

Reply:
I guess it isn’t always present (`Elmish_UseEffect_unknown_62445_114`, for example). The only way this could result in a collision is if someone is using code splitting plus JS modules. But maybe better to be on the safe / verbose side?

Reply:
A collision could happen if there are multiple JS files, and while code splitting is one way this could happen, it could also happen in other ways. One obvious example is during tests: we run tests in Node without bundling, so there are always multiple JS files. Another example is one of those trendy HTTP/2-powered bundle-less schemes, which might become more prevalent. I think it's safer to keep the file name in.

Are there any drawbacks to keeping it in?

Reply:
Makes sense 👍🏻 No real drawbacks except the component names are longer.

Reply:
Fixed. Here’s the component tree, now:

![image](https://user-images.githubusercontent.com/23478119/149199044-51d140cf-1db6-4f25-9dcd-f7367e5b4dd4.png)

Reply:
Well, considering it's intended for testing and debugging, I think more information is better. I would maybe cut off the `http://localhost:port` part if it's easy to do.

Reply:
No, the screenshot is bundled: the `http_localhost_8000_output_index_js` part is the file and the `output_Examples_UseEffectPrime_index_js` part is the function name.

And sure thing, I can remove everything but the file path 👍🏻

Reply:
Okay, here’s what it looks like now:

![image](https://user-images.githubusercontent.com/23478119/149209874-35537bf9-ac32-4357-b618-8737693bf3c3.png)

Reply:
Yeah, I think this is good. I think with a minimized bundle the function names would be non-informative, so it's good to have the file and the line/column.

Reply:
Oh good point about minimization 👍🏻